dashboard/app/controllers/followers_controller.rb

```diff
@@ -49,6 +67,41 @@ def remove
     )
   end
 
+  def remove_async
+    @section = Section.find_by_code params[:section_code]
+
+    if @section
+      f = Follower.where(section: @section.id, student_user_id: current_user.id).first
+    end
+
+    unless @section && f
+      render json: {
+        result: "error",
+        sections: current_user.sections_as_student.map(&:summarize)
+      }
+      return
+    end
+
+    @teacher = @section.user
+
+    authorize! :destroy, f
+    f.delete
```

Comment:
There's quite a bit of work here, and I believe we've been working to keep controllers thin and push work down into our models.  This might be a good place to do that - in particular, if the work of notifying a teacher is something we'd like to do any time a student-teacher association is destroyed, that work should live in the model (I think).